### ISSUE TITLE
Cannot build with AndroidX Hilt Work (Cannot find symbol class ApplicationComponent)

### ISSUE THREAD

**Issue by VincentJoshuaET** [google/dagger#2277]
Will we have to stay with 2.30.1 until AndroidX gets an update?

**Comment by VincentJoshuaET** [comment]
I hope there will be a comprehensive guide to migrate from the AndroidX libraries to the built in features.

**Comment by VincentJoshuaET** [comment]
Ok, here is how I did it:

```
class SampleWorker @AssistedInject constructor(
    @Assisted context: Context,
    @Assisted params: WorkerParameters,
    private val dependency: Dependency
) : CoroutineWorker(context, params) {

    override suspend fun doWork(): Result {
        TODO()
    }

}
```

```
@AssistedFactory
interface SampleWorkerFactory {
    fun create(context: Context, params: WorkerParameters): SampleWorker
}
```

```
class AndroidWorkerFactory @Inject constructor(private val factory: SampleWorkerFactory) : WorkerFactory() {

    override fun createWorker(context: Context, name: String, params: WorkerParameters): ListenableWorker {
        return factory.create(context, params)
    }

}
```

```
@HiltAndroidApp
class SampleApplication : Application(), Configuration.Provider {

    @Inject
    lateinit var factory: AndroidWorkerFactory

    override fun getWorkManagerConfiguration(): Configuration {
        return Configuration.Builder().setWorkerFactory(factory).build()
    }

}
```

**Comment by SteinerOk** [comment]
Have the same error.
It because ApplicationComponent deleted in new version of Dagger, but it used in AndroidX Hilt libraries.
We can just wait, wait...

**Comment by guillermomuntaner** [comment]
Same thing. Looking forward for some instructions from the Dagger/Hilt team and/or upgrades in `androidx.hilt:hilt-work`

**Comment by VincentJoshuaET** [comment]
> Same thing. Looking forward for some instructions from the Dagger/Hilt team and/or upgrades in `androidx.hilt:hilt-work`

Look into my third reply.

**Comment by danysantiago** [comment]
Sorry for the trouble, the `androidx.hilt` libraries are scheduled to be release on January 27th (if all goes well), but we had to release Dagger first with these changes. So please just hold on a bit before upgrading if you are using the Worker extension.

**Comment by shekibobo** [comment]
Is it not possible to delay removal of `ApplicationComponent` until the other internal libraries that depend on it were also released?

**Comment by jakoss** [comment]
> Is it not possible to delay removal of `ApplicationComponent` until the other internal libraries that depend on it were also released? 

My thoughts exactly. Even more - Google deprecated the 'ApplicationComponent' in last version for gradual migration and failed to migrate their own library (and we're talking about really minor change), which is baffling for me.
I know it's all in alpha, but such situation could be easily avoided

**Comment by shekibobo** [comment]
> > Is it not possible to delay removal of `ApplicationComponent` until the other internal libraries that depend on it were also released? 
> 
> My thoughts exactly. Even more - Google deprecated the 'ApplicationComponent' in last version for gradual migration and failed to migrate their own library (and we're talking about really minor change), which is baffling for me.
> I know it's all in alpha, but such situation could be easily avoided 

To be clear, the question wasn't a rhetorical complaint, but wondering if there could be an intermediate release by either library that could bridge the gap.

**Comment by bcorso** [comment]
Yes, I think that's a possibility. I can discuss with the team when we get back on Tuesday.

**Comment by hiyjeain** [comment]
Maybe we should have a migration guide and upgrade risk warning on dagger.dev website to decrease developer's debugging time.

**Comment by Chang-Eric** [comment]
Thanks for the feedback everyone. We've just released a new version 2.31.1 that adds back `ApplicationComponent`.
